lib/timers/promises.js

```diff
@@ -33,25 +32,21 @@ function cancelListenerHandler(clear, reject) {
 
 function setTimeout(after, value, options = {}) {
   const args = value !== undefined ? [value] : value;
-  if (options == null || typeof options !== 'object') {
-    return PromiseReject(
-      new ERR_INVALID_ARG_TYPE(
-        'options',
-        'Object',
-        options));
+  try {
+    validateObject(options, 'options');
+  } catch (err) {
+    return PromiseReject(err);
```

Comment:
You can make the function `async` instead.

Reply:
Marking these functions as `async` would prove the specific optimizations with promises down in the line useless as primordials like `SafePromisePrototypeFinally()`, `Promise()` and `PromiseReject()` are used.

However if one of the maintainers decide that we can remove the usage of those primordials and straight up make the function `async`, then this could be done.

cc @jasnell

Reply:
Sorry, line 95 was hidden in the diff.